lib/cli/config.js

```diff
@@ -45,36 +65,40 @@ const parsers = (exports.parsers = {
 /**
  * Loads and parses, based on file extension, a config file.
  * "JSON" files may have comments.
+ *
+ * @private
  * @param {string} filepath - Config file path to load
  * @returns {Object} Parsed config object
- * @private
  */
 exports.loadConfig = filepath => {
   let config = {};
   const ext = path.extname(filepath);
+
+  debug(`loadConfig: "${filepath}"`);
   try {
-    if (/\.ya?ml/.test(ext)) {
+    if (ext === '.yml' || ext === '.yaml') {
       config = parsers.yaml(filepath);
-    } else if (ext === '.js') {
-      config = parsers.js(filepath);
-    } else {
+    } else if (ext === '.json' || ext === '.jsonc') {
       config = parsers.json(filepath);
+    } else {
+      config = parsers.js(filepath);
```

Comment:
`require` will load both JS and JSON. Our documentation [states](https://next.mochajs.org/#priorities) JS is top priority. It unfortunately also [states](https://next.mochajs.org/#custom-locations) confusingly selects JSON is default; why?

It can be argued that the latter refers to _standard_ JSON, not the Mocha config file, so not breaking -- JSONC != JSON where this is concerned.

For a real breaking change, Mocha should _only_ allow comments in ".mocharc.jsonc".

Reply:
>  Our documentation states JS is top priority. It unfortunately also states confusingly JSON is default; why?

"Priority" refers to the order in which configuration files are merged, not the order in which they are searched for.  Sounds like that needs clarification.

> Mocha should only allow comments in ".mocharc.jsonc". We should deprecate use of comments in ".mocharc.json" and follow the standard.

IMO, this is unnecessarily hostile to the user; we don't lose anything by parsing `.json` files as JSONC, and we can't do that with `require()`.  Would appreciate other input on this.  @craigtaub @cspotcode @juerba etc?

So then my requested change is to revert the order so that the `else` block calls the `json` parser, and the previous `else if` block calls the `js` parser.

Reply:
> > Mocha should only allow comments in ".mocharc.jsonc". We should deprecate use of comments in ".mocharc.json" and follow the standard.
> 
> IMO, this is unnecessarily hostile to the user; we don't lose anything by parsing `.json` files as JSONC, and we can't do that with `require()`.

I objected to allowing comments to JSON when the PR was first introduced, though I realized the potential need for comments in a config file. Now we have that via JSONC. But I fail to find any value in adding app-specific behavior to a standard file format that renders it unusable anywhere else. Perfect example right here -- we're unable to load a JSON file via `require` because of our comment _feature_.

And I wouldn't call it user hostile either. I think it would match user expectation for a JSON file to be able to pass [JSONLint](https://jsonlint.com/) _and_ to be `require`able.

Reply:
> But I fail to find any value in adding app-specific behavior to a standard file format [...]

I agree, we should follow standards whenever possible, which is user- and maintainer-friendly. Expectations are clearly defined for both sides. 

So yes:
> We should deprecate use of comments in ".mocharc.json" and follow the standard.

Reply:
That means we _deprecate_ it instead of break it unless we want to cut v7.

Anybody who has a `.mocharc` (or specific config file lacking a file extension) containing JSON with comments will encounter an exception.

If this is how we want to go, then I'll live with it, but I can't get behind releasing this as-is in a minor.

Reply:
I'd like to see the following config-related changes in v7.0:
1. Remove comment stripping processing for ".mocharc.json".
1. Change the default for config pathname without known extension from JSON to just be `require`-able.
1. Deprecate module-relative _file_ processing fallback (unnecessary).
1. Allow for configuration by package.

Reply:
> That means we deprecate it instead of break it unless we want to cut v7.
> ...but I can't get behind releasing this as-is in a minor.

If "breaking change" was previously unclear, I meant semver-major.
Deprecation sometime between now and then...

Simple change in meantime would be to remove the comments from "example/config/.mocharc.json", which can be done semver-patch.